src/validators.ts

```diff
@@ -0,0 +1,159 @@
+import Types from './types';
+import { isObject } from './utils';
+import Constants from './constants';
+import { IdentityApiData } from '@mparticle/web-sdk';
+import { valueof } from './utils';
+
+type IdentityAPIMethod = 'login' | 'logout' | 'identify' | 'modify';
+
+type ValidationIdentitiesReturn = {
+    valid: boolean;
+    error?: valueof<typeof Constants.Messages.ValidationMessages>;
+};
+
+const Validators = {
+    isValidAttributeValue: function(value: any): boolean {
+        return value !== undefined && !isObject(value) && !Array.isArray(value);
+    },
+
+    // Neither null nor undefined can be a valid Key
+    isValidKeyValue: function(key: any): boolean {
+        return Boolean(key && !isObject(key) && !Array.isArray(key));
+    },
+
+    isStringOrNumber: function(value: any): boolean {
+        return typeof value === 'string' || typeof value === 'number';
+    },
+
+    isNumber: function(value: any): boolean {
+        return typeof value === 'number';
+    },
+
+    isFunction: (fn: any): boolean => {
+        return typeof fn === 'function';
+    },
+
+    validateIdentities: function(
+        identityApiData: IdentityApiData,
+        method?: IdentityAPIMethod
```

Comment:
Hmmm.  Seems like when looking at the function signature in isolation it is in fact optional.  When doing a global search for the usage in the code base it seems we are always passing a method in.

Trying to rack my brain on why, and it's _probably_ because there was some potential for different methods doing slightly different things, as opposed to it only being just special treatment for modify.  I think keeping the method as optional is fine then.